### ISSUE TITLE
Unparseable date

### ISSUE THREAD

**Issue by robertvazan** [jorabin/KeePassJava2#27]
I am getting this exception when parsing KDBX 3.1 file that was edited (loaded, modified, saved) by pykeepass:

```
Caused by: java.lang.IllegalStateException: java.text.ParseException: Unparseable date: "2021-01-02T03:31:49.386013+00:00"
	at org.linguafranca.pwdb.kdbx.Helpers.toDate(Helpers.java:83)
	at org.linguafranca.pwdb.kdbx.simple.converter.TimeConverter.read(TimeConverter.java:36)
	at org.linguafranca.pwdb.kdbx.simple.converter.TimeConverter.read(TimeConverter.java:29)
	...
	at org.linguafranca.pwdb.kdbx.simple.SimpleDatabase.load(SimpleDatabase.java:192)
	...
```

KeePassJava2 expects timestamps to match format `yyyy-MM-dd'T'HH:mm:ss'Z'`.

So which implementation of the format is wrong, pykeepass or KeePassJava2?

KeePassX opens the file without error and displays the timestamps correctly.

**Comment by jorabin** [comment]
Interesting, not seen before as far as I know, which means:

1) Keepass itself only generates timestamps of the format `yyyy-MM-dd'T'HH:mm:ss'Z'`
2) If Keepass itself opens such files without error then I guess other implementations should too, since that is the only measure by which one can measure conformance
3) as to which is wrong, I would call on the old dictum: "be strict in your production and be generous in your consumption" so pykeepass is wrong, but KeePassJava2 should (maybe) accommodate incorrect implementations

Please attach an example of such a file, without which it won't be possible to make progress on this. Better still, propose a fix.

Jo

**Comment by robertvazan** [comment]
Here's an example KDBX file that was created in KeePassX and then edited with pykeepass, which added single entry.

[timestamp_test.zip](https://github.com/jorabin/KeePassJava2/files/5771697/timestamp_test.zip)

Password is 'passwordless'.

**Comment by Evidlo** [comment]
This is the date format returned by Python's `date.isoformat()`.  While there is no Keepass spec, there is an unofficial one [here](https://github.com/Evidlo/keepassxc-specs/blob/master/kdbx-xml/rfc.md), which claims that datetimes should be in one of the ISO8601 formats.

**Comment by jorabin** [comment]
That's a nice piece of documentation. 

For many of the motivations discussed there I also created various pieces of [documentation](https://github.com/jorabin/KeePassJava2#keepassjava2-and-keepass) which you may have read in familiarising with this project. In particular I created [an XSD](https://github.com/jorabin/KeePassJava2/blob/master/KDBX.3.1.xsd) (XML Schema, a formal definition of XML in XML).

The format defined in that file says "This is ISO Date time stored as UTC (Z time zone)", which like the other definition is based on observation, rather than scrutiny of the C# code for KeePass, which hitherto has not been necessary.

I'll have a look at the file you attached in an earlier comment later.

**Comment by jorabin** [comment]
Hi @robertvazan 

Your file doesn't demonstrate the problem. Here's its XML: 
[scratch_1.xml.zip](https://github.com/jorabin/KeePassJava2/files/5793412/scratch_1.xml.zip)

All the timestamps are in the "correct" format, did you attach the wrong file?

**Comment by robertvazan** [comment]
Sorry. I didn't test the last file. This one definitely causes the exception:

[bogus-timestamp.zip](https://github.com/jorabin/KeePassJava2/files/5794793/bogus-timestamp.zip)

I am using KeePassJava2 2.1.4 from Maven Central and I am getting this exception:

```
Caused by: java.lang.IllegalStateException: java.text.ParseException: Unparseable date: "2021-01-11T09:18:56.769557+00:00"
	at org.linguafranca.pwdb.kdbx.Helpers.toDate(Helpers.java:83)
	at org.linguafranca.pwdb.kdbx.simple.converter.TimeConverter.read(TimeConverter.java:36)
	at org.linguafranca.pwdb.kdbx.simple.converter.TimeConverter.read(TimeConverter.java:29)
	at org.simpleframework.xml.convert.AnnotationStrategy.read(AnnotationStrategy.java:126)
	...
	at org.simpleframework.xml.core.Persister.read(Persister.java:444)
	at org.linguafranca.pwdb.kdbx.simple.SimpleDatabase.load(SimpleDatabase.java:192)
```

Password is "passwordless".

**Comment by jorabin** [comment]
Format allowed now looser, hope that's useful ...

**Comment by jorabin** [comment]
Closed in 2.2.1
